rasa/importers/importer.py

```diff
@@ -372,7 +376,14 @@ def _messages_from_user_utterance(event: UserUttered) -> Message:
 
 
 def _messages_from_action(event: ActionExecuted) -> Message:
-    return Message(event.e2e_text or "", data={MESSAGE_ACTION_NAME: event.action_name})
+    # we need to store the action text twice to be able to differentiate between user and bot text in NLU processing
```

Comment:
could you explain why? I think it should work without storing it twice as you don't add anymore information?

Reply:
This is changed [here](https://github.com/RasaHQ/rasa/blob/d3ed2e84f80bb3a967c2f4edd39e432ee6420f7c/rasa/importers/importer.py#L380-L384). We make sure that user text and action text are stored separately.